lib/Models/Terria.js

```diff
@@ -669,20 +691,45 @@ Terria.prototype.addInitSource = function(initSource) {
     runLater(function() {
       that.showSplitter = initSource.showSplitter;
     });
+  } else {
+    that.showSplitter = false;
+  }
+
+  if (fromStory === true) {
+    viewerChangeListener = this.afterViewerChanged.addEventListener(function() {
+      if (that.viewerMode === ViewerMode.Leaflet) {
+        that.leaflet.zoomTo(that.initialView, 0.0);
+      } else {
+        that.cesium.zoomTo(that.initialView, 0.0);
+      }
+    });
   }
 
   if (defined(initSource.viewerMode) && !defined(this.userProperties.map)) {
     var desiredMode = (initSource.viewerMode || "").toLowerCase();
+
     if (desiredMode === "2d") {
+      if (fromStory && this.viewerMode === ViewerMode.Leaflet) {
+        return;
+      }
       this.viewerMode = ViewerMode.Leaflet;
+      if (defined(viewerChangeListener)) viewerChangeListener();
```

Comment:
I'm not sure why the `afterViewerChanged` listener is being removed here (that's what calling `viewChangeListener()` does, right?). It seems backwards. If the viewer is already the right one, you don't need that `afterViewerChanged` listener, so removing it (or never installing it in the first place) makes sense. But if the viewer is changing, you need the listener in order to set the view after the change. And then I think you'd want the handler to remove itself from the Event, to avoid the bug I described above.

Reply:
I'm a bit confused. I'm not removing `viewerChangeListener` here? I changed the handler to remove itself after zoom.

Reply:
`viewerChangeListener` is assigned the return value of a Cesium `addEventListener`. So `viewerChangeListener()` removes the listener. Probably not what you expected. :)

Reply:
oh yeah i missed that. Thanks!